src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -67,4 +71,32 @@ def get(self, svc_id):
 
         daemon['rgw_metadata'] = metadata
         daemon['rgw_status'] = status
+
         return daemon
+
+
+@ApiController('rgw/proxy')
+@AuthRequired()
+class RgwProxy(RESTController):
+    @cherrypy.expose
+    def default(self, *vpath, **params):
+        try:
+            rgw_client = RgwClient.admin_instance()
+
+        except NoCredentialsException as e:
+            cherrypy.response.headers['Content-Type'] = 'application/json'
+            cherrypy.response.status = 401
+            return json.dumps({'message': e.message})
```

Comment:
in python 3 the `Exception` object does not have the `message` attribute. Use `str(e)` to get the message string.